src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/UnixPkcs12Reader.cs

```diff
@@ -40,13 +41,10 @@ protected void ParsePkcs12(ReadOnlySpan<byte> data)
                 // Windows compatibility: Ignore trailing data.
                 ReadOnlySpan<byte> encodedData = reader.PeekEncodedValue();
 
-                unsafe
-                {
-                    IntPtr tmpPtr = Marshal.AllocHGlobal(encodedData.Length);
-                    Span<byte> tmpSpan = new Span<byte>((byte*)tmpPtr, encodedData.Length);
-                    encodedData.CopyTo(tmpSpan);
-                    _tmpManager = new PointerMemoryManager<byte>((void*)tmpPtr, encodedData.Length);
-                }
+                _tmpMemoryHandle = SafeLocalAllocWithClearOnDisposeHandle.Create(encodedData.Length);
```

Comment:
```suggestion

                // We use managed buffer to avoid GC compaction clones of the private key material
                _tmpMemoryHandle = SafeLocalAllocWithClearOnDisposeHandle.Create(encodedData.Length);
```